app/src/main/java/me/saket/dank/urlparser/UrlParser.java

```diff
@@ -367,6 +369,22 @@ public static boolean isGooglePlayUrl(String urlHost, String uriPath) {
     return urlHost.endsWith("play.google.com") && uriPath.startsWith("/store");
   }
 
+  public static boolean isYouTubeUrl(String urlHost) {
+       return (urlHost.endsWith("youtube.com") || urlHost.endsWith("youtu.be"));
+  }
+
+  @Nullable
+  public static String getVideoIDFromYouTubeUrl(String url){
+    String videoId = null;
+    String regex = "http(?:s)?:\\/\\/(?:m.)?(?:www\\.)?youtu(?:\\.be\\/|be\\.com\\/(?:watch\\?(?:feature=youtu.be\\&)?v=|v\\/|embed\\/|user\\/(?:[\\w#]+\\/)+))([^&#?\\n]+)";
```

Comment:
I'm always paranoid of breaking regexes. Can we add a test for this to `UrlParserTest`?